applicationset/services/repo_service.go

```diff
@@ -31,15 +33,23 @@ type Repos interface {
 
 	// GetDirectories returns a list of directories (not files) within the target repo
 	GetDirectories(ctx context.Context, repoURL string, revision string) ([]string, error)
-}
 
-func NewArgoCDService(db db.ArgoDB, gitCredStore git.CredsStore, submoduleEnabled bool) Repos {
+	// Close any open connections
+	Close()
+}
 
+func NewArgoCDService(db db.ArgoDB, gitCredStore git.CredsStore, submoduleEnabled bool, repoClientset repoapiclient.Clientset) (Repos, error) {
+	closer, repoClient, err := repoClientset.NewRepoServerClient()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
From previous experience and a quick look at all the existing code that uses repo client (for eg. app controller) it seems the practice is to create the repoclient from the clientset when a request comes in so this should probably go into GetFiles and GetDirectories function.